cluster/gce/manifests/etcd.manifest

```diff
@@ -55,10 +55,12 @@
       }
         ],
     "livenessProbe": {
-      "httpGet": {
-        "host": "127.0.0.1",
-        "port": {{ port }},
-        "path": "/health"
+      "exec": {
+        "command": [
+                 "/bin/sh",
+                 "-c",
+                 "reply=$(curl -s -o /dev/null -w '%{http_code}' {{ etcd_livenessprobe_args }} {{ etcd_apiserver_protocol }}://127.0.0.1:{{ port }}/health);if [ \"$reply\" -eq 200 ]; then exit 1; fi; exit 0;"
```

Comment:
@timstclair This is what I did for the liveness probe as your suggested offline, PTAL. Thanks!

Reply:
One problem with this approach is it adds curl as a dependency to the etcd image. I don't know what else is in that image, but I imagine it should be capable of being a minimal image (which means less maintenance burden to keep it patched). What is the base image for etcd currently?

Reply:
I don't think we need any maintenance to keep it patched to work with curl. Base image for etcd in current k8s release is etcd v3.3.10. Did I answer your question?
https://github.com/kubernetes/kubernetes/blob/b6c8ef298ebfceb66422c416f55ab6076b7ff65d/cluster/images/etcd/Makefile#L29

Reply:
What I meant was, adding a dependency on shell & curl will prevent us from [moving this image to distroless/static](https://github.com/kubernetes/kubernetes/issues/70249), which means we need to expend more effort to keep this image patched.

Reply:
Ah, I read the issue you mentioned on moving images to distroless/static and talked to @yuwenma , yes the baseimage for etcd is busybox, which is exactly what you are moving from. So curl or shell dependency won't work in future. I think that means I have to do the side-car approach. Thanks!

Reply:
@timstclair and @yuwenma 
Actually I can use etcdctl to do endpoint health check, that should be fine?
```
wenjiazhang@kubernetes-master ~ $ ETCDCTL_API=3 /run/etcdctl endpoint health --endpoints=https://127.0.0.1:2379/health --cacert /etc/srv/kubernetes/pki/etcd-apiserver-ca.crt --cert /etc/srv/kubernetes/pki/etcd-apiserver-client.crt --key /etc/srv/kubernetes/pki/etcd-apiserver-client.key 
https://127.0.0.1:2379/health is healthy: successfully committed proposal: took = 1.551113ms
```
etctctl documentation: https://github.com/etcd-io/etcd/tree/v3.3.10/etcdctl#endpoint-health

Reply:
Awesome, sounds like that should work! etcdctl is packaged in the etcd image?

Reply:
Note that the exec probe status is based on the command exit status. Does etcdctl return a non-zero exit code for unhealthy endpoints?